subworkflows/local/generate_downstream_samplesheets/main.nf

```diff
@@ -0,0 +1,54 @@
+//
+// Subworkflow with functionality specific to the nf-core/mag pipeline
+//
+
+workflow SAMPLESHEET_MAG {
+    take:
+    ch_processed_reads
+
+    main:
+    format     = 'csv' // most common format in nf-core
+    format_sep = ','
+
+
+    ch_list_for_samplesheet = ch_processed_reads
+            .filter { meta, sample_id, instrument_platform,fastq_1,fastq_2,fasta -> (fastq_1 && fastq_2) && !fasta }.view()
+                .map {
+                        meta, sample_id, instrument_platform,fastq_1,fastq_2,fasta ->
+                            def sample        = meta.id
+                            def run           = meta.run_accession  //this should be optional
+                            def group         = ""
+                            def short_reads_1 = file(params.outdir).toString() + '/' + meta.id + '/' + fastq_1.getName()
+                            def short_reads_2 = file(params.outdir).toString() + '/' + meta.id + '/' + fastq_2.getName()
+                            def long_reads    = ""
+                [sample: sample, run: run, group: group, short_reads_1: short_reads_1, short_reads_2: short_reads_2, long_reads: long_reads]
+        }
+        .tap { ch_colnames }
+
+    channelToSamplesheet(ch_colnames, ch_list_for_samplesheet, 'downstream_samplesheets', format, format_sep)
```

Comment:
Joon has updated the function to also include the pipeline name as an additional argument, please check detaxier for the example (this will be helpful when @LilyAnderssonLee 's differentialabundance PR comes in :)